### PR TITLE
Volumes should be listed when attached to execs

### PR DESCRIPTION
**Pull request by mmacheerpuppy**
This PR set ensures that session volumes are listed with `unweave ls`

### CODE REVIEW

File: services/execsrv/exec.go [hunk 1]
```diff
@@ -13,11 +13,13 @@ var (
 )
 
 type Store interface {
+	AddSharedVolumes(exec types.Exec) error
 	Create(project string, exec types.Exec) error
 	Get(id string) (types.Exec, error)
 	GetDriver(id string) (string, error)
 	List(filterProject *string, filterProvider *types.Provider, filterActive bool) ([]types.Exec, error)
 	Delete(project, id string) error
+	RemoveSharedVolumes(exec types.Exec) error
```
Comment (noorvir) [inline services/execsrv/exec.go:22]: All volumes are shared so I don't think this needs a prefix. I actually think this shouldn't be a new method. This should be done inside the `Create` and `Delete` methods. That's the only time an `ExecVolume` is altered.

File: db/queries/exec_volumes.sql [hunk 2]
```diff
@@ -0,0 +1,10 @@
+-- name: ExecVolumeGet :many
+select exec_id, volume_id, mount_path from unweave.exec_volume as ev where ev.exec_id = $1;
+
+-- name: ExecVolumeCreate :exec
+insert into unweave.exec_volume (exec_id, volume_id, mount_path)
+values ($1, $2, $3) on conflict do nothing;
```
Comment (noorvir) [inline db/queries/exec_volumes.sql:6]: Not sure about the `conflict do nothing` part. We should probably throw an error so this doesn't do something unexpected. 

Also I realized the unique constraint is execID and volumeID. This means you can't mount the same volume twice at two different mount_paths on the same exec. That seems an unnecessary constraint. 

So we probably want to update the constraint to be on all three columns and fail on conflict.

File: services/execsrv/service.go [hunk 3]
```diff
@@ -217,10 +221,13 @@ func (s *ExecService) Terminate(ctx context.Context, id string) error {
...
 		return fmt.Errorf("failed to terminate exec: %w", err)
 	}
-
 	if err = s.store.UpdateStatus(exec.ID, types.StatusTerminated); err != nil {
 		return fmt.Errorf("failed to update exec status in store: %w", err)
 	}
+	err = s.store.RemoveSharedVolumes(exec)
+	if err != nil {
+		return fmt.Errorf("failed to delete shared volumes in store: %w", err)
+	}
```
Comment (noorvir) [inline services/execsrv/service.go:230]: This should go inside the store Delete method

File: db/queries/exec_volumes.sql (hunk 2 above)
Comment (mmacheerpuppy) [reply db/queries/exec_volumes.sql:6]: I looked into this and the uniqueness constraint is on execID, volumeID, and mount path

File: services/execsrv/store.go [hunk 4]
```diff
@@ -157,15 +169,29 @@ func (p postgresStore) List(projectID *string, filterProvider *types.Provider, f
...
+		res[idx] = dbExecToExec(exec, dbExecVolumesToVolumes(volumes), dbSSHKeyToSSHKey(keys))
 	}
 
 	return res, nil
 }
 
-func (p postgresStore) Delete(project, id string) error {
-	//TODO implement me
-	panic("implement me")
+func (p postgresStore) Delete(_, id string) error {
```
Comment (noorvir) [inline services/execsrv/store.go:183]: Let's delete project from interface definition if not used

**Review by noorvir — APPROVED** [review]
Looks good! Left a comment about removing updating the store interface but not critical

File: services/execsrv/store.go (hunk 4 above)
Comment (mmacheerpuppy) [reply services/execsrv/store.go:183]: No worries just wasn't sure if we wanted to keep this around for any reason

**Review by noorvir — APPROVED** [review]

**Pull request merged**